### PR TITLE
Simplify data sharing handler 

### PR DESCRIPTION
**Pull request by dannygoldstein**
This PR is depends on #1314. The current data sharing page lists all groups as potential targets for data sharing, but an API exception is thrown if a user tries to share data with a group they are not a member of.  This PR removes the requirement that a user be a member of the target group for data sharing, fixing this issue.

### CODE REVIEW

**Comment by dannygoldstein** [comment]
Failures unrelated

File: skyportal/handlers/api/sharing.py [hunk 1]
```diff
@@ -51,42 +51,76 @@ def post(self):
...
-        groups = Group.query.filter(Group.id.in_(group_ids))
-        if not all([group in self.current_user.accessible_groups for group in groups]):
-            return self.error(
-                "Insufficient permissions: you must have access to each "
-                "target group you wish to share data with."
-            )
-        obj_id = None
+
+        ok_groups = Group.query.filter(Group.id.in_(group_ids))
+        ok_group_ids = [og.id for og in ok_groups]
```
Comment (acrellin) [inline skyportal/handlers/api/sharing.py:58]:
```suggestion
        valid_groups = Group.query.filter(Group.id.in_(group_ids))
        valid_group_ids = [g.id for g in valid_groups]
```

File: skyportal/handlers/api/sharing.py [hunk 2]
```diff
@@ -51,42 +51,76 @@ def post(self):
...
+        groups = ok_groups
+
+        phot_obj_ids = []
+        spec_obj_ids = []
+
         if phot_ids:
-            query = Photometry.query.filter(Photometry.id.in_(phot_ids))
-            for phot in query:
+            ok_phot = Photometry.query.filter(Photometry.id.in_(phot_ids))
+            ok_phot_ids = [op.id for op in ok_phot]
```
Comment (acrellin) [inline skyportal/handlers/api/sharing.py:70]: same comment here (ok -> valid)
ok feels ambiguous to me as I can't tell whether it's ok from the perspective of a valid ID or is accessible to user

File: skyportal/handlers/api/sharing.py [hunk 3]
```diff
@@ -51,42 +51,76 @@ def post(self):
...
                 # Grab obj_id for use in websocket message below
-                if obj_id is None:
-                    obj_id = phot.obj_id
+                phot_obj_ids.append(phot.obj_id)
+
         if spec_ids:
-            query = Spectrum.query.filter(Spectrum.id.in_(spec_ids))
-            for spec in query:
+            ok_spec = Spectrum.query.filter(Spectrum.id.in_(spec_ids))
+            ok_spec_ids = [os.id for os in ok_spec]
```
Comment (acrellin) [inline skyportal/handlers/api/sharing.py:92]: ok -> valid

**Review by acrellin — APPROVED** [review]

**Review by dmitryduev — APPROVED** [review]
LGTM! Perhaps worth merging in the latest master.

**Comment by dannygoldstein** [comment]
@dmitryduev  Fixed merge conflicts

**Pull request merged**